www/js/diary/current.js

```diff
@@ -0,0 +1,224 @@
+ angular.module('emission.main.diary.current', ['ui-leaflet',
+                                                'ngCordova', 
+                                                'emission.services', 
+                                                'ionic'])
+
+.controller('CurrMapCtrl', function($scope, Config, $state, $timeout, $ionicActionSheet,leafletData, Logger, $window) {
+    
+  console.log("controller CurrMapCtrl called from current.js");
+  var db = window.cordova.plugins.BEMUserCache;
+  MANUAL_INCIDENT = "manual/incident";
+  BACKGROUND_LOCATION = "background/location";
+  $scope.mapCtrl = {}; 
+  var directions = ["N", "NE", "E", "SE", "S", "SW", "W", "NW"]; 
+
+  $scope.diary = function() {
+    console.log('diary', "in diary");
+    $state.go("root.main.diary");
+  };   
+
+  angular.extend($scope.mapCtrl, {
+    defaults: {
+    }
+  });
+
+  angular.extend($scope.mapCtrl.defaults, Config.getMapTiles());
+
+  console.log("mapCtrl", $scope.mapCtrl);
+
+  var date = new Date();
+  var time = date.toTimeString().split(' ')[0].split(':');
+
+  var startTimeFn = function(ts) {
+    var date = new Date(ts*1000);
+    var hours = date.getHours();
+    var minutes = date.getMinutes();
+    var amOrPm = hours < 12 ? 'AM' : 'PM';
+    hours = hours % 12;
+    hours = hours ? hours : 12;
+    minutes = minutes < 10 ? '0'+ minutes : minutes;
+    $scope.startTime = hours + ':' + minutes + ' ' + amOrPm;
+  };
+
+  var msToMph = function(ms) {
```

Comment:
Maybe offer people a choice between miles and km? Ideally, when people customize the UI, they would change this - a US-only one would use mph and the rest of the world would use kmph. But if you feel strongly about this, and it is not too much work, you can have it be controlled by a slider in the profile.